Calculator/ViewModel/CalculatorOperationsProtocol.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+enum Operation {
+    case Addition
+    case Subtraction
+    case Multiplication
+    case Division
+}
+
+protocol CalculatorOperationsProtocol {
+    func operate(opt1: Double, opt2: Double, operation: Operation) -> Double
+}
+
+extension CalculatorOperationsProtocol {
+
+    func operate(opt1: Double, opt2: Double, operation: Operation) -> Double {
+
+        switch operation {
+        case .Addition:
```

Comment:
so here, an option could be that perhaps `Operation` could be a protocol, and then an implementation could be `Addition`, then you don't need this switch statement and could add new types of operation without changing this code (which then conforms to the open/closed principle). What do you think?